### PR TITLE
Updates to casacore package

### PR DESCRIPTION
**Pull request by mpokorny**
My first Spack PR. I tried to meet the guidelines, but it is unclear whether you'd prefer this broken into multiple PRs, one per commit. Happy to resubmit, if needed.

Note that I have contributed to the upstream casacore package in the past, and I'm currently working on fixes to issues in the casacore build for which I've put in place workarounds in this PR. I hope to be able to follow up with another PR for the casacore package once the upstream is fixed.

### CODE REVIEW

File: var/spack/repos/builtin/packages/casacore/package.py [hunk 1]
```diff
@@ -12,26 +12,39 @@ class Casacore(CMakePackage):
...
     depends_on('bison', type='build')
     depends_on('blas')
     depends_on('lapack')
     depends_on('cfitsio@3.181:')
-    depends_on('wcslib@4.20:')
+    depends_on('wcslib@4.20:+cfitsio')
     depends_on('fftw~mpi@3.0.0:', when='+fftw')
-    depends_on('sofa-c', when='+sofa')
+    # SOFA dependency suffers the same problem in CMakeLists.txt as readline;
+    # however, as SOFA is far less likely to be present on most system, and as
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/casacore/package.py]:
```suggestion
    # however, as SOFA is far less likely to be present on most systems, and as
```

File: var/spack/repos/builtin/packages/casacore/package.py [hunk 2]
```diff
@@ -12,26 +12,39 @@ class Casacore(CMakePackage):
     homepage = "https://github.com/casacore/casacore"
     url      = "https://github.com/casacore/casacore/archive/v2.4.1.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/casacore/package.py]: Would you like to add yourself as a maintainer? Just add:
```python
maintainers = ['mpokorny']
```
That way, if someone has trouble building the `casacore` package, or wants to submit a PR to update it, we can ping you to take a look.

File: var/spack/repos/builtin/packages/casacore/package.py (hunk 1 above)
Comment (mpokorny) [reply var/spack/repos/builtin/packages/casacore/package.py]: I'll take care of it

File: var/spack/repos/builtin/packages/casacore/package.py (hunk 2 above)
Comment (mpokorny) [reply var/spack/repos/builtin/packages/casacore/package.py]: Sure, I'll do that

**Pull request merged**